### PR TITLE
Add support for 16-bit WAV sound lumps

### LINKED ISSUE #101: audio: support for 16-bit WAV sound lumps
I'd like to add support for SFX lumps in WAV format, just as the implementation @SmileTheory contributed to Crispy Doom.

### PR DESCRIPTION
**Pull request by rfomin**
Based on Crispy Doom. Tested it with Doom Sound Bulb [[doomworld]](https://www.doomworld.com/forum/topic/110822-doom-sound-bulb-hd-sounds-that-stay-true-to-the-original/).

Fixes #101

### CODE REVIEW

File: Source/i_sound.c [hunk 1]
```diff
@@ -266,7 +327,7 @@ static boolean addsfx(sfxinfo_t *sfx, int channel, int pitch)
       sfx_alen *= 4;
 
       // haleyjd 06/03/06: don't need original lump data any more
-      Z_ChangeTag(data, PU_CACHE);
+      //Z_ChangeTag(data, PU_CACHE);
```
Comment (rfomin) [inline Source/i_sound.c:330]: I don't know why this doesn't work.

File: Source/i_sound.c [hunk 2]
```diff
@@ -186,33 +187,77 @@ static boolean addsfx(sfxinfo_t *sfx, int channel, int pitch)
...
+         if (samplelen > lumplen - 44)
+            samplelen = lumplen - 44;
+
+         bits = data[34] | (data[35] << 8);
+
+         // Reject non 8 or 16 bit
+         if (bits != 16 && bits != 8)
+            return false;
+
+         data += 44 - 8;
```
Comment (fabiangreffrath) [inline Source/i_sound.c:231]: You increase the `data` pointer here, that's why it fails to free it later on.

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: Source/i_sound.c (hunk 2 above)
Comment (rfomin) [reply Source/i_sound.c]: Fixed in https://github.com/fabiangreffrath/woof/pull/122/commits/eb7fbf9eaea1ca6a8e213ea9a5d44d33abaa9567

File: Source/i_sound.c [hunk 3]
```diff
@@ -133,7 +133,7 @@ static void stopchan(int handle)
    channelinfo[handle].id = NULL;
 }
 
-#define SOUNDHDRSIZE 8
+static int SOUNDHDRSIZE;
```
Comment (fabiangreffrath) [inline Source/i_sound.c:136]: Please initialize to 8.

**Comment by fabiangreffrath** [comment]
Beautiful, thank you!

**Pull request merged**